app/frontend/components/pages/dashboard/dashboardPage.tsx

```diff
@@ -22,201 +22,210 @@ import AccountsDashboard from '../accounts/accountsDashboard'
 import {State} from '../../../state'
 import PoolOwner from '../advanced/poolOwner'
 import ErrorModals from './errorModals'
-
-interface Props {
-  selectedMainTab: any
-  selectMainTab?: (value: string) => void
-}
+import {useState} from 'preact/hooks'
+import {SubTabs, MainTabs} from '../../../constants'
 
 const StakingPage = () => {
+  const subTabs = [SubTabs.DELEGATE_ADA, SubTabs.CURRENT_DELEGATION, SubTabs.STAKING_HISTORY]
+  const defaultSubTab = SubTabs.DELEGATE_ADA
+  const mainSubTab = SubTabs.SHELLEY_BALANCES
```

Comment:
subTabs, defaultSubTab and mainSubTab and defined in all of these pages separately, although these variables are shared among them. I would therefore extract them to props and pass their values at their instantiation, so the sub and defaults are in one place and it would be more readable

Reply:
If I understand you correctly, you would prefer the `DashboardMobile` being called only once, and instead of passing the values from its parent component, we would pass them like `subtabs={MainTabs[selectedMainTab].subtabs}` `defaultSubTab={MainTabs[selectedMainTab].defaultSubTab}` etc. If this is what you mean, I previously planned to do it this way, but I found it a lot less readable to create this massive object in which for every tab I had to define a list of subtabs, defaultSubTab, and mainTab even thou as you pointed out, the binding happens multiple times in that case. 

It's just a matter of creating a big object with subtabs, defaultSubTab, and mainTab for every MainTab or defining these in the respective components, in which the second choice seemed a lot more readable to me, that's it.

Reply:
Sure but that way it's all in one place and it would be better typed, plus the exact same code
```
<DashboardMobileContent
          subTabs={subTabs}
          defaultSubTab={defaultSubTab}
          mainSubTab={mainSubTab}
        />
```
appear 3 times in the code, which doesn't seem like a good refactoring direction to me

Reply:
The fact that some code appears 3 times is a bad smell for sure, but not necessarily a bad decision. First of all, it doesn't really make sense to me to detach the logic of what is displayed on each main tab to some object, secondly this has other implications in terms of UX. I decided to remove that duplicate code as you pointed out and implement it like this.

```
const DashboardMobileContent = ({activeMainTab}: DashboardMobileProps) => {
  const [activeSubTab, setActiveSubTab] = useState(MainPages[activeMainTab].defaultSubTab)
  const subTabs = MainPages[activeMainTab].subTabs
  const mainSubTab = MainPages[activeMainTab].mainSubTab
  useEffect(
    () => {
      setActiveSubTab(MainPages[activeMainTab].defaultSubTab)
    },
    [activeMainTab]
  )
  ```
 I had to use `useEffect` to rerender the default sub tab if the main tab changes, which is terms of UX pretty terrible, (causes the tab to change right await, and it's content only on the second render which is quite noticeable that instead of the content changing all at once, it changes basically twice.)
 
So in this case, it is more than beneficial to have a mobile view not as one component, but to have a mobile view for every tab, even though it happens to be a bit of duplicate code in this example.
 
 I also decided to add a hook for viewport size so we don't render mobile and desktop view at once and I ll be probably reverting to the previous approach even tho it might make a bigger mess out of it.

Reply:
I pushed two versions, if you want to, pull it and check out it behaves.